dist/tools/headerguards/headerguards.py

```diff
@@ -68,16 +71,18 @@ def fix_headerguard(filename):
 
     tmp.seek(0)
     if guard_found == 3:
-        for line in difflib.unified_diff(inlines, tmp.readlines(), "%s" % filename, "%s" % filename):
+        for line in difflib.unified_diff(inlines, tmp.readlines(),
+                                         "%s" % filename, "%s" % filename):
             sys.stdout.write(line)
     else:
         print("%s: no / broken header guard" % filename, file=sys.stderr)
         return False
 
-if __name__=="__main__":
+
+if __name__ == "__main__":
     error = False
     for filename in sys.argv[1:]:
-        if fix_headerguard(filename) == False:
```

Comment:
this changes semantics

Reply:
indeed, so `fix_headerguard` should return True at the end ?

Reply:
```fix_headerguard()``` returns "False" on error. The call site checks for that. Why change?

Reply:
>  Why change?

```flake8 --config=dist/tools/flake8/flake8.cfg dist/tools/headerguards/
...
dist/tools/headerguards/headerguards.py:80:38: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
```
I chose the second solution, because it's cleaner: `fix_headerguard()` should explicitly return True when problems are detected, False otherwise.

Reply:
Then why not ```if fix_headerguard() is False:```?



> it's cleaner: fix_headerguard() should explicitly return True when problems are detected, False otherwise.

debatable.

Reply:
in general [explicit is better than implicit](https://www.python.org/dev/peps/pep-0020/#id3). With the first suggestion, the function returns None when problems are detected, False otherwise. Testing between False and None could lead to confusion and buggy code. That's why I chose the other proposition.

Reply:
The code was explicit before you broke it. And why returning True on error leads to less confusion than returning False is something I don't think you'll find in any style guide.

The function *could* return True when there's no error. Would that improve readability? I doubt it.

Reply:
Let's not spend too much time on this and move forward. I've just updated the PR with your preference.

Reply:
Its basically a function that would benefit from being split in parts and the error case done with an exception, but its another PR.